applications/unit_tests/helpers.hpp

```diff
@@ -41,11 +76,43 @@ namespace detail
         return instance;
     }
 
-    inline llri::Adapter* selectAdapter(llri::Instance* instance)
+    template<typename Func, typename ...Args>
+    inline void iterateAdapters(llri::Instance* instance, Func&& function, Args&&... args)
     {
         std::vector<llri::Adapter*> adapters;
         REQUIRE_EQ(instance->enumerateAdapters(&adapters), llri::result::Success);
-        return adapters[0];
+        REQUIRE_UNARY(adapters.size() > 0);
+        
+        constexpr size_t adapterStr = nameHash(LLRI_SELECTED_TEST_ADAPTERS);
+        constexpr size_t all = nameHash("All");
+        
+        std::vector<llri::Adapter*> selectedAdapters;
+        
+        if constexpr (adapterStr == all)
+        {
+            selectedAdapters = adapters;
+        }
+        else
+        {
+            auto selectedAdapterNames = split(LLRI_SELECTED_TEST_ADAPTERS, ';');
+            
+            for (auto* adapter : adapters)
+            {
+                llri::adapter_info info = adapter->queryInfo();
+                
+                if (std::find(selectedAdapterNames.begin(), selectedAdapterNames.end(), info.adapterName) != selectedAdapterNames.end())
```

Comment:
depending on how long and how different these names are, all these string compare calls could be either fast or slow? maybe something more consistent like hashing all the selected names once, and hashing the name of each adapter and doing hash comparisons might be benificial?

Reply:
I have no consistent way of caching information (other than a disk file) because command-line unit tests run the executable multiple times meaning stuff like statics are cleared each time

Reply:
I could hash them once though yeah

Reply:
oh i meant hashing them once before the adapter loop XD

Reply:
makes sense :) I'll do that